File: editorial/chap06/answer_B41.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
 
int X, Y;
 
int main() {
	// 入力
	cin >> X >> Y;
 
	// 逆から考えていく
	vector<pair<int, int>> Answer;
	while (X >= 2 || Y >= 2) {
		Answer.push_back(make_pair(X, Y));
		if (X > Y) X -= Y;
		else Y -= X;
	}
	reverse(Answer.begin(), Answer.end());
 
	// 出力
	cout << Answer.size() << endl;
	for (int i = 0; i < Answer.size(); i++) {
		cout << Answer[i].first << " " << Answer[i].second << endl;
	}
	return 0;
}

File: editorial/chap06/python/answer_B41.py
# 入力
X, Y = map(int, input().split())

# 逆から考えていく
Answer = []
while X>=2 or Y>=2:
	Answer.append([X, Y])
	if X > Y:
		X -= Y
	else:
		Y -= X
Answer.reverse()

# 出力
print(len(Answer))
for i in range(len(Answer)):
	print(str(Answer[i][0]) + ' ' + str(Answer[i][1]))
